src/libraries/System.Private.Xml/src/System/Xml/Core/XmlWriterAsync.cs

```diff
@@ -599,5 +599,21 @@ private async Task WriteLocalNamespacesAsync(XPathNavigator nsNav)
                 await WriteAttributeStringAsync("xmlns", prefix, XmlReservedNs.NsXmlNs, ns).ConfigureAwait(false);
             }
         }
+
+        public async ValueTask DisposeAsync()
+        {
+            await DisposeAsyncCore().ConfigureAwait(false);
+            Dispose(false);
```

Comment:
not sure I follow here, why do we Dispose after DisposeAsync?

Reply:
This is by the pattern we are following. By my understanding as we suppressing finalize we are calling Dispose(false) which is called by finalizer.